client/lib/plans/index.js

```diff
@@ -204,3 +215,33 @@ export function plansLink( url, site, intervalType ) {
 
 	return url;
 }
+
+export function applyTestFiltersToPlansList( planName ) {
+	const filteredPlanConstantObj = Object.assign( {}, plansList[ planName ] );
+	let filteredPlanFeaturesConstantObj = getPlanFeaturesConstantObj( plansList[ planName ].getFeatures() );
+
+	if ( ! isWordadsInstantActivationEnabled() ) {
+		filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
+			( _, key ) => key !== WORDADS_INSTANT
+		);
+	} else if ( planName === PLAN_PREMIUM ) {
+		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithWordAdsInstantActivation;
+	}
+
+	if ( ! isGoogleVouchersEnabled() ) {
+		filteredPlanFeaturesConstantObj = pickBy( filteredPlanFeaturesConstantObj,
+			( _, key ) => key !== FEATURE_GOOGLE_AD_CREDITS
+		);
+	} else if ( planName === PLAN_PREMIUM ) {
+		filteredPlanConstantObj.getDescription = plansList[ planName ].getDescriptionWithGoogleVouchers;
```

Comment:
Could we maybe add some abtest checks in the actual `getDescription` method rather than overriding it here? I think it would be easier to follow what's happening.